ocaml/xapi/xapi_task.ml

```diff
@@ -48,3 +48,10 @@ let cancel ~__context ~task =
     Db.Task.set_current_operations ~__context ~self:task ~value:[(Ref.string_of (Context.get_task_id __context)), `cancel];
 	if not(Xapi_xenops.task_cancel ~__context ~self:task) && not(Storage_access.task_cancel ~__context ~self:task)
 	then info "Task.cancel is falling back to polling"
+
+let set_status ~__context ~self ~status =
+	if not (Db.is_valid_ref __context self) then
+		raise (Api_errors.Server_error("Invalid ref", [Ref.string_of self]))
```

Comment:
The first argument to `Server_error` should be one of the pre-defined strings in api_errors.ml - these are errors understood by XenCenter and the CLI which can be converted into readable error messages.